scripts/pre_commit_linter.py

```diff
@@ -1478,6 +1511,88 @@ def _check_directive_scope(all_files, parsed_js_files):
     return summary_messages
 
 
+def _check_js_component_count(all_files, parsed_js_files):
```

Comment:
The name of the function matches just one half of its purpose, it doesn't say anything about the name check.

Reply:
Done